maintenance_account/models/account_move.py

```diff
@@ -19,7 +19,8 @@ def unlink(self):
 
     def action_post(self):
         super().action_post()
-        equipment_model = self.env["maintenance.equipment"]
+        # Prevent error if user does not have permission to create equipments
+        equipment_model = self.env["maintenance.equipment"].sudo()
```

Comment:
Pass also company of the invoice for multi-company environments, with context `force_company`. You need to do it per document.